src/memkind_local.c

```diff
@@ -6,26 +6,41 @@
 #include <memkind/internal/memkind_mem_attributes.h>
 #include <memkind/internal/memkind_private.h>
 
-struct hi_cap_loc_numanodes_t {
+struct loc_numanodes_t {
     int init_err;
     struct bitmask **per_cpu_numa_nodes;
 };
 
-static struct hi_cap_loc_numanodes_t
-    memkind_hi_cap_loc_numanodes_g[NODE_VARIANT_MAX];
+static inline void local_kind_finalize(const struct loc_numanodes_t *g)
+{
+    if (MEMKIND_LIKELY(!g->init_err)) {
+        if (g->per_cpu_numa_nodes) {
+            int num_cpus = numa_num_configured_cpus();
+            int i;
```

Comment:
Move this definition into the loop, you're not using this variable anywhere else.